src/tsdb.c

```diff
@@ -87,6 +91,21 @@ int SeriesAddSample(Series *series, api_timestamp_t timestamp, double value) {
     return TSDB_OK;
 }
 
+int SeriesUpdateSample(Series *series, api_timestamp_t timestamp, double value) {
+    Sample tmpSample;
```

Comment:
This functions needs to:
1. check if this chunk supports in-place updates
2. Using the series Iterator for this is dangerous because it might change and break this function.

Reply:
1. Why wouldn't it support in-place updates? are you talking about compressed chunks?
2. I wanted to keep the code DRY... I can write my own logic for iterating it, but it'll just bloat the code with duplicate logic, isn't it?

Reply:
1. yes
2. there's two options:
 a. Write a chunk iterator
 b. write a loop over the chunk, and later on we will refactor this into an iterator.